product/interfaces/templatefunctions/findProducts.go

```diff
@@ -33,25 +34,31 @@ func (tf *FindProducts) Func(ctx context.Context) interface{} {
 
 	/*
 		widgetName - used to namespace widget - in case we need pagination
-		config - map with certain keys - used to specifiy th searchRequest better
+		config - map with certain keys - used to specify the searchRequest better
 	*/
-	return func(namespace string, configs ...map[string]string) *application.SearchResult {
-		var searchConfig, keyValueFilters, filterConstrains map[string]string
+	return func(namespace string, configs ...pugjs.Map) *application.SearchResult {
+		var searchConfig, filterConstrains map[string]string
+		var keyValueFilters map[string][]string
 
 		if len(configs) > 0 {
-			searchConfig = configs[0]
+			searchConfig = configs[0].AsStringMap()
 		} else {
 			searchConfig = make(map[string]string)
 		}
 
+		keyValueFilters = make(map[string][]string)
 		if len(configs) > 1 {
-			keyValueFilters = configs[1]
-		} else {
-			keyValueFilters = make(map[string]string)
+			for key, value := range configs[1].AsStringIfaceMap() {
+				if str, ok := value.(pugjs.String); ok { // allow string
+					keyValueFilters[key] = []string{str.String()}
+				} else if arrStr, ok := value.([]string); ok { // and array of strings
+					keyValueFilters[key] = arrStr
+				}
+			}
 		}
 
 		if len(configs) > 2 {
-			filterConstrains = configs[2]
+			filterConstrains = configs[2].AsStringMap()
 		} else {
 			filterConstrains = make(map[string]string)
 		}
```

Comment:
Same as first comment, I would suggest to remove the else case and init filterConstraints at the start of the function